overlord/bomb.rb

```
@@ -0,0 +1,73 @@
+class Bomb
+  attr_accessor :status, :deactivation_attempts, :activation_code, :deactivation_code
+
+  ACTIVATION_CODE = '1234'
+  DEACTIVATION_CODE = '0000'
+
+  def initialize(activation_code = ACTIVATION_CODE, deactivation_code = DEACTIVATION_CODE)
+    @status = :inactive
+    @deactivation_attempts = 0
+
+    @activation_code = correct_format?(activation_code) ? activation_code : ACTIVATION_CODE
+    @deactivation_code = correct_format?(deactivation_code) ? deactivation_code : DEACTIVATION_CODE
```

/me ninjas in

I don't dig this line. If I provide a poorly formatted deactivation code (say I fat-finger and give you too many digits), you silently set the code to a default value without warning me. Since I thought I set it to something else, I will of course keep trying that code until the bomb explodes in my face.

Better to throw an error and refuse to start up, than to silently do something unexpected and explodey. :bomb: :cry: